spectrochempy/analysis/fast_ica.py

```diff
@@ -0,0 +1,350 @@
+    of the underelying sources and a mixing matrix :math:`A`\ .
+
+    In terms of matrix equation:
+
+    .. math:: X = \bar{X} + A \cdot S^t + E
+
+    where :math:`\bar{X}` is the mean of the dataset and :math:`E` is the matrix of
+    residuals.
+
+    Parameters
+    ----------
+    %(AnalysisConfigurable.parameters)s
+
+    See Also
+    --------
+    %(DecompositionAnalysis.see_also.no_FastICA)s
+    """
+    )
+
+    # ----------------------------------------------------------------------------------
+    # Runtime Parameters,
+    # only those specific to FastICA, the other being defined in AnalysisConfigurable.
+    # ----------------------------------------------------------------------------------
+    # define here only the variable that you use in fit or transform functions
+    _fastica = tr.Instance(
+        decomposition.FastICA,
+        help="The instance of sklearn.decomposition.FastICA used in this model",
+    )
+
+    # ----------------------------------------------------------------------------------
+    # Configuration parameters
+    # ----------------------------------------------------------------------------------
+
+    n_components = tr.Integer(
+        default_value=None,
+        allow_none=True,
+        help="Number of components (sources) to use.",
+    ).tag(config=True)
+
+    algorithm = tr.Enum(
+        ["parallel", "deflation"],
+        default_value="parallel",
+        help=("""Specify which algorithm to use for FastICA."""),
+    ).tag(config=True)
+
+    whiten = tr.Union(
+        (tr.Enum(["arbitrary-variance", "unit-variance"]), tr.Bool(False)),
+        default_value="arbitrary-variance",
+        allow_none=True,
+        help=(
+            """Specify the whitening strategy to use.
+
+- ``"arbitrary-variance"``\ : a whitening with variance arbitrary is used.
+- "unit-variance" : the whitening matrix is rescaled to ensure that each recovered
+  source has unit variance.
+- False : the data is already considered to be whitened, and no whitening is performed.
+"""
+        ),
+    ).tag(config=True)
+
+    fun = tr.Union(
+        (tr.Enum(["logcosh", "exp", "cube"]), tr.Callable(), tr.Unicode()),
+        default_value="logcosh",
+        allow_none=True,
+        help=(
+            r"""The functional form of the function used in the approximation to neg-entropy.
+
+- ``string``\ : could be either ``"logcosh"``\ , ``"exp"``\ , or ``"cube"``\ .
+- ``callable``\ : You can provide your own function. It should return a tuple containing
+  the value of the function, and of its derivative, in the point. The derivative should
+  be averaged along its last dimension.
+
+    .. code-block::
+
+        def my_g(x):
+            return x ** 3, (3 * x ** 2).mean(axis=-1)
+
+    .. note::
+        ``fun`` can be also a serialized function created using dill and base64
+        python libraries. Normally not used directly, it is here for internal
+        process. """
```

Comment:
Please dedent the ..code-block and ..note to the left margin. better looking as the column is already quite narrow in the html display.